apis/core/lib/domain/dimension-metadata/DimensionMetadata.ts

```diff
@@ -0,0 +1,46 @@
+import { Constructor } from '@tpluscode/rdfine'
+import { CsvMapping, DimensionMetadataCollection } from '@cube-creator/model'
+import { ColumnMapping } from '@cube-creator/model/ColumnMapping'
+import * as DimensionMetadata from '@cube-creator/model/DimensionMetadata'
+import { ResourceStore } from '../../ResourceStore'
+import * as id from '../identifiers'
+import { cc } from '@cube-creator/core/namespace'
+import { schema } from '@tpluscode/rdf-ns-builders'
+
+interface CreateColumnMetadata {
+  store: ResourceStore
+  csvMapping: CsvMapping
+  columnMapping: ColumnMapping
+}
+
+interface ApiDimensionMetadataCollection {
+  addDimensionMetadata(params: CreateColumnMetadata): DimensionMetadata.DimensionMetadata
+}
+
+declare module '@cube-creator/model' {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface DimensionMetadataCollection extends ApiDimensionMetadataCollection {
+  }
+}
+
+export default function Mixin<Base extends Constructor<DimensionMetadataCollection>>(Resource: Base) {
+  return class Impl extends Resource implements ApiDimensionMetadataCollection {
+    addDimensionMetadata(params: CreateColumnMetadata): DimensionMetadata.DimensionMetadata {
+      const identifier = params.csvMapping.createIdentifier(params.columnMapping.targetProperty)
+
+      if (this.hasPart && this.hasPart.some((dimMeta) => dimMeta.about === identifier)) {
+        throw new Error(`Dimension Metadata with identifier ${identifier.value} already exists`)
+      }
+
+      const dimensionMetadata = DimensionMetadata.create(this.pointer.node(id.dimensionMetadata(this, params.columnMapping)), {
```

Comment:
I've drifted a little away from really needing the `identifiers` module. In this case for example, `id.dimensionMetadata` will likely only ever be used here. 

I propose to move into this module instead